chain/network/src/peer_manager.rs

```diff
@@ -1467,122 +1647,120 @@ impl Handler<NetworkRequests> for PeerManagerActor {
+                                requested_edges: Default::default(),
+                                done: true,
+                            }),
+                        });
+                    } else {
+                        if ibf_msg.ibf_level == MIN_IBF_LEVEL {
+                            self.routing_table.add_peer(peer_id.clone(), ibf_set.clone());
+                            let ibf_set = self.routing_table.get_ibf_set(&peer_id).unwrap();
+                            ibf_set.lock().unwrap().set_seed(ibf_msg.seed);
+                        }
 
-                                            })
-                                            .collect();
+                        // Edges to send to the other side
+                        let (mut edges_for_peer, _) = self
+                            .routing_table
+                            .split_edges_for_peer(&peer_id, &ibf_msg.requested_edges);
 
-                                        // Add accounts to the routing table.
-                                        if !accounts.is_empty() {
-                                            debug!(target: "network", "{:?} Received new accounts: {:?}", act.config.account_id, accounts);
-                                        }
-                                        for account in accounts.iter() {
-                                            act.routing_table.add_account(account.clone());
-                                        }
+                        // Edges to add
+                        self.verify_and_broadcast_edges(
+                            ctx,
+                            peer_id.clone(),
+                            ibf_msg.edges,
+                            Default::default(),
+                        );
 
-                                        let new_data = SyncData { edges: new_edges, accounts };
+                        // Combine data from ibfs and decide what to do
 
-                                        if !new_data.is_empty() {
-                                            act.broadcast_message(
-                                                ctx,
-                                                SendMessage { message: PeerMessage::RoutingTableSync(new_data) },
-                                            )
-                                        };
+                        if !ibf_msg.done
+                            && ibf_msg.ibf_level >= MIN_IBF_LEVEL
+                            && ibf_msg.ibf_level <= MAX_IBF_LEVEL
+                        {
+                            let (known_edges, unknown_edge_hashes, unknown_edges_count) =
+                                self.routing_table.exchange_routing_tables_using_ibf(
+                                    &peer_id,
+                                    ibf_set.clone(),
+                                    ibf_msg.ibf_level as usize,
+                                    &ibf_msg.ibf,
+                                    ibf_msg.seed,
```

Comment:
If the other peer sends a different seed will make crash our node rn. In general we should ban the peer if it sends different seed.

Reply:
Fixed